jaworski_cellseg/jwski_widget.py

```diff
@@ -19,47 +21,89 @@ def __init__(self, napari_viewer: napari.viewer.Viewer, parent=None):
-            count_widget.native,
-            region_selection_widget.native,
-            inferer_widget,
+            self.config_widget.native,
+            self.data_widget.native,
+            self.pre_process_data_widget.native,
+            self.count_widget.native,
+            self.region_selection_widget.native,
+            self.inferer_widget,
         ]:
             layout.addWidget(widget)
 
     def configure_inferer_widget(self, inferer_widget):
         """
         Configure default settings for the inference widget.
         """
+        # Get configuration
+
         # Model selection and inference settings
         inferer_widget.model_choice.setCurrentIndex(1)  # Select SwinUNetR
-        inferer_widget.use_window_choice.setChecked(True)
-        inferer_widget.model_input_size.setValue(96)
-        inferer_widget.window_overlap_slider.setValue(50)
-        inferer_widget.thresholding_checkbox.setChecked(True)
+        inferer_widget.use_window_choice.setChecked(
+            self.current_config["inference_use_window_choice"]
+        )
+        inferer_widget.window_size_choice.setCurrentIndex(
+            inferer_widget.window_size_choice.findText(
+                str(self.current_config["inference_window_size"])))
+        
+        inferer_widget.model_input_size.setValue(
+           self.current_config["inference_input"]
+        )
+        inferer_widget.window_overlap_slider.setValue(
+            self.current_config["inference_overlap"]
+        )
+        inferer_widget.thresholding_checkbox.setChecked(
+            self.current_config["inference_perform_threhold"]
+        )
 
         # Set probability threshold (0.6) and instance segmentation settings
-        inferer_widget.thresholding_slider.setValue(60)
-        inferer_widget.use_instance_choice.setChecked(True)
+        inferer_widget.thresholding_slider.setValue(
+            self.current_config["inference_perform_threhold_value"]
+        )
+        inferer_widget.use_instance_choice.setChecked(
+            self.current_config["inference_instance_segmentation"]
+        )
 
         # Configure instance segmentation with Voronoi-Otsu method parameters
-        voronoi_widget = inferer_widget.instance_widgets.methods["Voronoi-Otsu"]
-        voronoi_widget.counters[0].setValue(2.5)
-        voronoi_widget.counters[2].setValue(25.00)
+        voronoi_widget = inferer_widget.instance_widgets.methods[
+            self.current_config["inference_instance_segmentation_option"]
+        ]
+        voronoi_widget.counters[0].setValue(
+            self.current_config["inference_instance_segmentation_spot_signma"]
+        )
+        voronoi_widget.counters[1].setValue(
+            self.current_config["inference_instance_segmentation_outline_signma"]
+        )
+        voronoi_widget.counters[2].setValue(
+            self.current_config[
+                "inference_instance_segmentation_small_object_removal"
+            ]
+        )
```

Comment:
This workflow reappears! Consider adding a helper function and reuse.

Reply:
I applied your first suggestion